### ISSUE TITLE
Retrain AST-based reranker with correct metrics and labels

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#712]
## Problem

Current reranker HURTS accuracy (-6.7%). Two root causes:

### 1. Wrong Training Metric
Trained on **MRR (Mean Reciprocal Rank)** which tests:
- "Is the relevant document at position 1 or 2 or 3?"

Should train on **Precision/Recall** which tests:
- "Can you separate relevant from irrelevant documents?"

MRR optimizes for position, not relevance discrimination.

### 2. Potentially Wrong Labels
Training data labels might be incorrect:
- Labels based on keyword matching
- May not reflect actual answer presence
- Need evaluation-informed labels

### New Training Approach

**1. Better metric:**
```python
# OLD: MRR loss
loss = -log(1 / rank_of_relevant_doc)

# NEW: Binary cross-entropy for relevance
loss = BCE(predicted_relevance, true_relevance)
```

**2. Better labels:**
- Use evaluation results to label which retrieved sentences actually helped
- Positive: Sentences that led to correct answer
- Negative: Sentences that didn't help or led to wrong answer

**3. Keep architecture:**
- AST-based cross-attention (this is good)
- Focus on fixing loss function and labels

### AST-First Approach

The reranker already uses AST structure for cross-attention. Keep this - it's the right approach. Just fix the training.

## Expected Impact

Fix ranking quality issue (currently BM25 alone performs better than BM25+reranker)

## Effort

1 week:
- Days 1-2: Generate new training data with evaluation-informed labels
- Days 3-4: Implement new loss function, retrain
- Day 5: Evaluate and iterate

## Dependencies

- Should be done AFTER:
  - Disabling current reranker (#708)
  - BM25 analysis (#709)
  - Object verification (#710)
  - Query expansion model (#711)
- This is medium-term improvement, not blocking immediate progress

## Related

Part of QA improvement epic. Final learned component after query expansion model.
